rcl/src/rcl/node.c

```diff
@@ -363,14 +362,16 @@ rcl_node_fini(rcl_node_t * node)
 }
 
 bool
-rcl_node_is_valid(const rcl_node_t * node)
+rcl_node_is_valid(const rcl_node_t * node, const rcl_allocator_t * allocator)
 {
-  RCL_CHECK_ARGUMENT_FOR_NULL(node, false, rcl_get_default_allocator());
+  rcl_allocator_t alloc = allocator ? *allocator : rcl_get_default_allocator();
+  RCL_CHECK_ARGUMENT_FOR_NULL(node, false, alloc);
+  RCL_CHECK_ALLOCATOR_WITH_MSG(allocator, "invalid allocator", return false);
```

Comment:
Sorry that was unclear in my previous comment: I meant that we should check that `alloc` is valid _before_ using it in `RCL_CHECK_ARGUMENT_FOR_NULL`:
something like:
```c++
rcl_allocator_t alloc = allocator ? *allocator : rcl_get_default_allocator();
RCL_CHECK_ALLOCATOR_WITH_MSG(alloc, "invalid allocator", return false);
RCL_CHECK_ARGUMENT_FOR_NULL(node, false, alloc);
```